geode-redis/src/integrationTest/java/org/apache/geode/redis/SetsIntegrationTest.java

```diff
@@ -98,6 +101,42 @@ public void testSAddSCard() {
     assertThat(jedis.scard(key)).isEqualTo(strings.size());
   }
 
+  @Rule
+  public ExpectedException exceptionRule = ExpectedException.none();
+
+  @Test
+  public void testSAdd_withExistingKey_ofWrongType_shouldReturnError() {
+    String key = "key";
+    String stringValue = "preexistingValue";
+    String[] setValue = new String[1];
+    setValue[0] = "set value that should never get added";
+
+    exceptionRule.expect(JedisDataException.class);
+    exceptionRule.expectMessage(RedisConstants.ERROR_WRONG_TYPE);
+
+    jedis.set(key, stringValue);
+    jedis.sadd(key, setValue);
+  }
+
+  @Test
+  public void testSAdd_withExistingKey_ofWrongType_shouldNotOverWriteExistingKey() {
+    String key = "key";
+    String stringValue = "preexistingValue";
+    String[] setValue = new String[1];
+    setValue[0] = "set value that should never get added";
+
+    jedis.set(key, stringValue);
+
+    try {
+      jedis.sadd(key, setValue);
+    } catch (Exception exception) {
```

Comment:
It seems like you are expecting a certain exception here. Can it be made more specific than "Exception"?

Reply:
done!